packages/components/src/InputDate/InputDate.test.tsx

```diff
@@ -180,9 +173,44 @@ it("displays the calendar when input is focused with a click", () => {
   );
   const input = getByDisplayValue(date);
 
-  act(() => {
-    fireEvent.click(input);
-  });
+  fireEvent.click(input);
 
   expect(getByText("15")).toBeInTheDocument();
 });
+
+describe("when InputDate is used within a Modal", () => {
```

Comment:
this is definitely more of an integration test than purely a unit test, but I can't think of another effective way to verify this behavior since in isolation a test asserting that on ESC press the picker closes would pass even without this PR

Reply:
I guess, this would also work on just the normal datepicker that's triggered by a button. So maybe this test should still be in the DatePicker test. 

Alternatively, you are testing that it shouldn't trigger the escape on elements outside of the DatePicker. So you could test it like so

```tsx
const handleEscape = jest.fn()
const handleKeyDown = e => e.key === "Escape" && handleEscape()
window.addEventListener("keydown", handleKeyDown )
...

<div onKeyDown={handleKeyDown}>
  <DatePicker />
</div>

...

expect(handleEscape).not.toHaveBeenCalled()
```

Reply:
correct. this will also apply to the base DatePicker. I think it's simply more likely to have an `InputDate` inside a `Modal` than a `DatePicker` just because of what the components do, but it's entirely possible that you _could_  have it nested in something else listening to ESC.

I'm fine with adding another one like your example in DatePicker, but I do want to have at least 1 test that specifically covers the case we are seeking to address to begin with!

Reply:
hmmm I just noticed it auto focuses the picker upon opening for the base DatePicker, so my line of `nextMonthButton.focus();` in the test isn't necessary

even without that line it passes. happy to remove it if we think it's doing more harm than good.

Reply:
let's do it on a different PR. this one be good!